Source/WebCore/Modules/mediastream/MediaStreamTrackProcessor.cpp

```diff
@@ -71,10 +71,11 @@ MediaStreamTrackProcessor::~MediaStreamTrackProcessor()
 ExceptionOr<Ref<ReadableStream>> MediaStreamTrackProcessor::readable(JSC::JSGlobalObject& globalObject)
 {
     if (!m_readable) {
-        m_readableStreamSource = makeUniqueWithoutRefCountedCheck<Source>(m_track->privateTrack(), *this);
+        if (!m_readableStreamSource)
+            lazyInitialize(m_readableStreamSource, makeUniqueWithoutRefCountedCheck<Source>(m_track->privateTrack(), *this));
         auto readableOrException = ReadableStream::create(*JSC::jsCast<JSDOMGlobalObject*>(&globalObject), *m_readableStreamSource);
         if (readableOrException.hasException()) {
-            m_readableStreamSource = nullptr;
```

Comment:
The fixes to this file are me applying Youenn's fix (https://github.com/WebKit/WebKit/commit/4ab2d90be17370ffef2c89ac3eefeeb906ad2987) which should get cherry-picked to main any day now. I'll rebase my PR if the cherry-pick happens first.